### PR TITLE
Create an AdminUserService (BRIDGE-3267)

### PR DESCRIPTION
**Pull request by alxdarksage**
The integration tests can no longer use combined admin/participant accounts, so these have had to be split apart. Some other clean up once I had to create more accounts to use our most recent approach to cleaning up accounts.

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AccountsTest.java [hunk 1]
```diff
@@ -130,7 +130,8 @@ public void crudAccount() throws Exception {
                 .languages(ImmutableList.of("en", "fr"))
                 .password(PASSWORD)
                 .note("test note 1")
-                .clientTimeZone("America/Los_Angeles");
+                .clientTimeZone("America/Los_Angeles")
+                .orgMembership(orgAdmin.getSession().getOrgMembership());
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/AccountsTest.java:134]: Why is this necessary? It shouldn't be.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SessionTest.java [hunk 2]
```diff
@@ -31,138 +33,131 @@
 import com.google.common.collect.Lists;
 
 public class SessionTest {
+    
+    private TestUser user;
+    private TestUser researcher;
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/SessionTest.java:38]: Just double check that one account didn't work here.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/Exporter3Test.java [hunk 3]
```diff
@@ -419,10 +421,10 @@ public void accountWithRoleHasNoVersions() throws Exception {
...
 
         // Add test_user data group.
-        ParticipantsApi participantsApi = admin.getClient(ParticipantsApi.class);
-        StudyParticipant participant = participantsApi.getParticipantById(userId, false).execute().body();
-        participant.addDataGroupsItem("test_user");
-        participantsApi.updateParticipant(userId, participant).execute();
+        AccountsApi accountsApi = admin.getClient(AccountsApi.class);
+        Account account = accountsApi.getAccount(userId).execute().body();
+        account.addDataGroupsItem("test_user");
+        accountsApi.updateAccount(userId, account).execute();
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/Exporter3Test.java:427]: You cannot update the developer account through updateParticipant, so these APIs were switched over to update via the accounts API.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/OrganizationTest.java [hunk 4]
```diff
@@ -197,20 +196,30 @@ public void testMembership() throws Exception {
...
-        StudyParticipant participant = participantsApi.getParticipantById(user.getUserId(), false).execute().body();
-        participant.setOrgMembership(orgId2);
-        participantsApi.updateParticipant(user.getUserId(), participant).execute();
+        // cannot change organizational affiliation on an update. We need an admin to retrieve the account because
+        // it's moving between organizations
+        AccountsApi adminAccountsApi = admin.getClient(AccountsApi.class);
+        Account account = adminAccountsApi.getAccount(user.getUserId()).execute().body();
+        
+        account.setOrgMembership(orgId2);
+        adminAccountsApi.updateAccount(user.getUserId(), account).execute();
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/OrganizationTest.java:205]: Cannot update a developer account through the participant APIs, so moved this over to use the accounts API.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SessionTest.java [hunk 5]
```diff
@@ -31,138 +33,131 @@
...
+                info.getEnrolledOn().isBefore(info.getWithdrawnOn()));
+        assertNull(info.isEnrolledBySelf());
+        assertTrue(info.isWithdrawnBySelf()); // see above
     }
     
     @Test
     public void canGetStudyParticipantWithAllData() throws Exception {
-        TestUser user = TestUserHelper.createAndSignInUser(SessionTest.class, true);
-        try {
-            List<Role> roles = ImmutableList.of(ADMIN);
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/SessionTest.java:91]: This test user account was also an admin, so they had to be split or the participant could not be updated through the participant APIs.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SessionTest.java [hunk 6]
```diff
@@ -31,138 +33,131 @@
...
+        assertEquals(AccountStatus.ENABLED, asUpdated.getStatus());
     }
 
     @Test
     public void sessionUpdatedWhenResearcherUpdatesOwnAccount() throws Exception {
-        TestUser researcher = TestUserHelper.createAndSignInUser(SessionTest.class, false, Role.RESEARCHER);
-        try {
-            // Roles. ADMIN should not be set.
-            List<Role> roles = Lists.newArrayList(Role.RESEARCHER, Role.ADMIN);
-            List<String> dataGroups = Lists.newArrayList("group1");
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/SessionTest.java:132]: Same, this test participant also had roles and had to be split.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/UserParticipantTest.java [hunk 7]
```diff
@@ -62,90 +67,85 @@ public void after() throws Exception {
...
     
     @Test
     public void canUpdateDataGroups() throws Exception {
+        user = TestUserHelper.createAndSignInUser(getClass(), true);
+                
         // Developer in this test is not a test user.
         List<String> dataGroups = ImmutableList.of("sdk-int-1", "sdk-int-2");
 
-        ForConsentedUsersApi usersApi = developer.getClient(ForConsentedUsersApi.class);
+        ForConsentedUsersApi usersApi = user.getClient(ForConsentedUsersApi.class);
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/UserParticipantTest.java:141]: Besides some cleanup above, the changes here are again, to separate out a target participant account from the account that has the roles.

**Review by jnelsonjava — APPROVED** [review]
